Group3 feature hint

Khobaka opened the pull request:
Changes to Puzzle.cpp
- added structure for hint
- changes to DisplayBoardHints() function

Changes to Hangman-cpp
- changed hint condition (actually unnecessary)

aricbrillantes committed:
Initial Commit (added basic letter removal)

> Puzzle.cpp
- added displayBoardHint() function
-- currently removes ALL letters that are not in the answer; needs fixing

> hangman.cpp
- added hint input
-- if user types hint, displayBoardHint() will be called

Khobaka committed:
fixed hint function(I think)

changed to Puzzle.cpp
- added stucture hint
-changes in DIsplayBoardHInt() function

aricbrillantes committed:
fixed indentations

fixed indentations and brackets

aricbrillantes committed:
added hint limit

added numberOfHints variable in hangman.cpp (currently 3 hints per game)

aricbrillantes committed:
added comments

added comments for hint function and variables

aricbrillantes committed:
updated hint

- added messages when using hint
- players can no longer use a hint twice in one word

Khobaka committed:
CHEAT CODE TRY

- lots of changes to PUZZLE.cpp and Hangman.cpp

Khobaka committed:
Cheatcode revamp

fixed some weird bug

aricbrillantes committed:
merged cheat code functions with revised code

Khobaka committed:
added a Cheatcode

Gain a life

Khobaka committed:
little shits

ajctan reviewed (dismissed):
spamming directional buttons doesn't work while using cheat. naming conventions are non-existent. print statements on the same line as others. gitgudkid.

aricbrillantes committed:
minor fixes

fixed:
- hints resetting
- line spacing (?)

aricbrillantes committed:
added hint earning

- players will now earn 1 hint after every 5 wins

- removed "Default letter 's' used." message from SChecker.cpp

KlaudiaGaiaBorromeo reviewed (changes requested):
amount of removed letters in hint makes it impossible to lose in certain cases

aricbrillantes committed:
updated cheats (PRE CHECK-IN)

- added fixes to cheat function

EGCruz reviewed (commented):
gj

ajctan commented:
:/
